tools/workspace-plugin/src/generators/vrt-generator/index.ts

```diff
@@ -0,0 +1,126 @@
+import {
+  addProjectConfiguration,
+  joinPathFragments,
+  formatFiles,
+  generateFiles,
+  Tree,
+  offsetFromRoot,
+} from '@nx/devkit';
+import * as path from 'path';
+import { VisualRegressionSchema } from './schema';
+import { getProjectConfig, isPackageConverged } from '../../utils';
+import { addCodeowner } from '../add-codeowners';
+
+interface NormalizedSchema extends ReturnType<typeof normalizeOptions> {}
+
+export default async function (tree: Tree, schema: VisualRegressionSchema) {
+  const options = normalizeOptions(tree, schema);
+
+  addFiles(tree, options);
+
+  addCodeowner(tree, { packageName: options.projectConfig.name as string, owner: schema.owner as string });
+
+  await formatFiles(tree);
+}
+
+function addFiles(tree: Tree, options: NormalizedSchema) {
+  const root = options.projectConfig.root as string;
+
+  const templateOptions = {
+    ...options,
+    rootOffset: offsetFromRoot(root),
+    npmScope: options.workspaceConfig.npmScope,
+    componentName: options.project,
+    tmpl: '',
+  };
+
+  generateFiles(tree, path.join(__dirname, 'files'), path.join(root), templateOptions);
+}
+
+function normalizeOptions(tree: Tree, options: VisualRegressionSchema) {
+  const projectName = `${options.project}-visual-regression`;
```

Comment:
we need to filter out that user doesnt specify by mistake lets say a `react-text-stories` project or an application